scipy/sparse/tests/test_arithmetic1d.py

```diff
@@ -0,0 +1,337 @@
+"""Test of 1D arithmetic operations"""
+
+import pytest
+
+import numpy as np
+
+from scipy.sparse import coo_array, csr_array
+from scipy.sparse._sputils import isscalarlike
+
+
+spcreators = [coo_array, csr_array]
+math_dtypes = [np.int64, np.float64, np.complex128]
+
+
+def toarray(a):
+    if isinstance(a, np.ndarray) or isscalarlike(a):
+        return a
+    return a.toarray()
+
+@pytest.fixture
+def dat1d():
+    return np.array([3, 0, 1, 0], 'd')
+
+
+@pytest.fixture
+def datsp_math_dtypes(dat1d):
+    dat_dtypes = {dtype: dat1d.astype(dtype) for dtype in math_dtypes}
+    return {
+        sp: [(dtype, dat, sp(dat)) for dtype, dat in dat_dtypes.items()]
+        for sp in spcreators
+    }
+
+
+@pytest.mark.parametrize("spcreator", spcreators)
+class TestArithmetic1D:
+    def test_empty_arithmetic(self, spcreator):
+        shape = (5,)
+        for mytype in [
+            np.dtype('int32'),
+            np.dtype('float32'),
+            np.dtype('float64'),
+            np.dtype('complex64'),
+            np.dtype('complex128'),
+        ]:
+            a = spcreator(shape, dtype=mytype)
+            b = a + a
+            c = 2 * a
+            assert isinstance(a @ a.tocsr(), np.ndarray)
+            assert isinstance(a @ a.tocoo(), np.ndarray)
+            for m in [a, b, c]:
+                assert m @ m == a.toarray() @ a.toarray()
+                assert np.array_equal(m.dtype, mytype)
+                assert np.array_equal(toarray(m).dtype, mytype)
+
+    def test_abs(self, spcreator):
+        A = np.array([-1, 0, 17, 0, -5, 0, 1, -4, 0, 0, 0, 0], 'd')
+        assert np.array_equal(abs(A), abs(spcreator(A)).toarray())
+
+    def test_round(self, spcreator):
+        A = np.array([-1.35, 0.56, 17.25, -5.98], 'd')
+        Asp = spcreator(A)
+        assert np.array_equal(np.around(A, decimals=1), round(Asp, ndigits=1).toarray())
+
+    def test_elementwise_power(self, spcreator):
+        A = np.array([-4, -3, -2, -1, 0, 1, 2, 3, 4], 'd')
+        Asp = spcreator(A)
+        assert np.array_equal(np.power(A, 2), Asp.power(2).toarray())
+
+        # element-wise power function needs a scalar power
+        with pytest.raises(NotImplementedError, match='input is not scalar'):
+            spcreator(A).power(A)
+
+    def test_real(self, spcreator):
+        D = np.array([1 + 3j, 2 - 4j])
+        A = spcreator(D)
+        assert np.array_equal(A.real.toarray(), D.real)
+
+    def test_imag(self, spcreator):
+        D = np.array([1 + 3j, 2 - 4j])
+        A = spcreator(D)
+        assert np.array_equal(A.imag.toarray(), D.imag)
```

Comment:
for these and other similar cases I'd say `numpy.testing` functions are the right way to go yeah.